### DISCUSSION TITLE
Are the BAP able to get the following information

### DISCUSSION THREAD

**Discussion by flamecopper** [BinaryAnalysisPlatform/bap#1437] [General]
Is the file an executable?
Is the file a binary?
What architecture is the file compile on.
What format is the binary in
Is the binary stripped of the symbol table
Will there be a running process with the binary?
What was the original programming language used?
Just to name a few.

**Comment by ivg** [comment]
> Is the file an executable?

Yes, e.g., `bap specification /bin/ls | grep is-executable`

> Is the file a binary?

Well, we only work with binaries.

> What architecture is the file compile on.

It won't be able to tell on what architecture the binary was compiled on, but will be able to tell *to* what architecture it was compiled, e.g., `bap specification /bin/ls | grep arch`

> What format is the binary in

`bap specification /bin/ls | grep file-format`

> Is the binary stripped of the symbol table

Mm, it depends, `bap specification /bin/ls | grep named-symbol` will show the symbol table information. It will be non-empty for typical /bin/ls as it will show both static and dynamic symbols. And /bin/ls is a dynamic binary with only static table removed. If you will fully strip it, you will not see any symbols.

> What was the original programming language used?

Nope. There's no analysis right now that will be able to discover this.

> Just to name a few.

You may also use
```
bap dependencies /bin/ls
```
To get the list of imported and exported symbols as well as the required libraries. With `bap dependencies --recursive /bin/ls` you will get this information recursively. 


P.S. sorry for the late reply, I totally forgot that we have discussions. Thanks for the good question, by the way!
